### ISSUE TITLE
App crashes on login

### ISSUE THREAD

**Issue by qoqobolo** [status-im/status-react#15023]
# Bug Report

## Problem 

Managed to catch crashes in several cases.
This is the most reliable way to reproduce it:

1. UserA sends a contact request to UserB via `Chat screen` > `+` button > `Add a new contact`
2. UserB accepts the request
3. UserB performs log out > log in several times 

Logs for this case: [crashlog_112.log](https://github.com/status-im/status-mobile/files/10677795/crashlog_112.log)

Also, sometimes the app crashes for the sender of the contact request (step 3, in this case, will be: **UserA** performs log out > log in several times).

Another scenario that was caught in e2e in [this PR](https://github.com/status-im/status-mobile/pull/14988):
1. UserA sends a contact request to UserB via `Chat screen` > `+` button > `Add a new contact`
2. UserB accepts the request and just waits on the home screen

It did not reproduce after rerunning the tests. 

Most often the crash happens with this error:

![Screenshot 2023-02-07 at 18 28 58](https://user-images.githubusercontent.com/67952253/217319792-82a4735f-be5f-4dab-a357-fbdd7c891889.png)

However, sometimes there is no error at all (Crash on login as well, logs for this case: [geth.log](https://github.com/status-im/status-mobile/files/10677834/geth.log)) or another error in shown (unfortunately I don't have a screenshot, but you can likely see it in the first logs)

### Additional Information

- Status version: nightly 7/02
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

**Comment by pavloburykh** [comment]
> Another scenario that was caught in e2e in https://github.com/status-im/status-mobile/pull/14988:
UserA sends a contact request to UserB via Chat screen > + button > Add a new contact
UserB accepts the request and just waits on the home screen

Confirming from my side. Reproduced several times in this [PR](https://github.com/status-im/status-mobile/pull/14853#issuecomment-1420839055). 

On video below crash happens on 2:21

https://user-images.githubusercontent.com/97245802/217324833-29e9d935-6bcc-43be-a872-13e50275005b.mp4

**Comment by cammellos** [comment]
The relevant logs are:
```
02-07 17:26:55.653 17782     0 E Go      : goroutine 64940 [running]:
02-07 17:26:55.653 17782     0 E Go      : github.com/status-im/status-go/vendor/github.com/waku-org/go-discover/discover.(*UDPv5).SetFallbackNodes(0x0, {0x400577eac0?, 0x1?, 0x1?})
02-07 17:26:55.653 17782     0 E Go      : 	github.com/status-im/status-go/vendor/github.com/waku-org/go-discover/discover/v5_udp.go:851 +0x24
02-07 17:26:55.653 17782     0 E Go      : github.com/status-im/status-go/vendor/github.com/waku-org/go-waku/waku/v2/discv5.(*DiscoveryV5).SetBootnodes(...)
02-07 17:26:55.653 17782     0 E Go      : 	github.com/status-im/status-go/vendor/github.com/waku-org/go-waku/waku/v2/discv5/discover.go:183
02-07 17:26:55.654 17782     0 E Go      : github.com/status-im/status-go/vendor/github.com/waku-org/go-waku/waku/v2/node.(*WakuNode).SetDiscV5Bootnodes(0x7c7ab9189b?, {0x400577eac0?, 0x7c7cc247a0?, 0x400577eb00?})
02-07 17:26:55.654 17782     0 E Go      : 	github.com/status-im/status-go/vendor/github.com/waku-org/go-waku/waku/v2/node/wakunode2.go:706 +0x88
02-07 17:26:55.654 17782     0 E Go      : github.com/status-im/status-go/wakuv2.(*Waku).restartDiscV5(0x400605c000)
02-07 17:26:55.654 17782     0 E Go      : 	github.com/status-im/status-go/wakuv2/waku.go:1464 +0x198
02-07 17:26:55.654 17782     0 E Go      : github.com/status-im/status-go/wakuv2.(*Waku).seedBootnodesForDiscV5(0x400605c000)
02-07 17:26:55.654 17782     0 E Go      : 	github.com/status-im/status-go/wakuv2/waku.go:1436 +0x3d0
02-07 17:26:55.654 17782     0 E Go      : created by github.com/status-im/status-go/wakuv2.(*Waku).Start
02-07 17:26:55.654 17782     0 E Go      : 	github.com/status-im/status-go/wakuv2/waku.go:1206 +0x128

```

**Comment by cammellos** [comment]
@richard-ramos looks like it might be a race condition where we set the listener to `nil`

**Comment by cammellos** [comment]
@qoqobolo @pavloburykh this is fixed here
https://github.com/status-im/status-mobile/pull/14985

I can split it in a separate PR if necessary, but I don't mind having more eyes on that one :)

**Comment by pavloburykh** [comment]
> I can split it in a separate PR if necessary, but I don't mind having more eyes on that one :)

Thanx a lot for quick fix! Let's have it in one PR without split @qoqobolo are you ok with that? Will be able to take a look tomorrow morning.

**Comment by cammellos** [comment]
> > I can split it in a separate PR if necessary, but I don't mind having more eyes on that one :)
> 
> Thanx a lot for quick fix! Let's have it in one PR without split @qoqobolo are you ok with that? Will be able to take a look tomorrow morning.

ah another thing, it should not be related to contact requests, just logging in/out should trigger the issue

**Comment by qoqobolo** [comment]
> ah another thing, it should not be related to contact requests, just logging in/out should trigger the issue

Ah okay! We tried to reproduce it without contacts/requests in the account, but for some reason, it did work with only at least one contact. It might be a coincidence though. Or a crash for some other reason :) Will observe the behavior and e2e results. 

> I can split it in a separate PR if necessary, but I don't mind having more eyes on that one :)

> Thanx a lot for quick fix! Let's have it in one PR without split @qoqobolo are you ok with that?

Sure, let's test it all together in one PR, I'll join the testing tomorrow.
